### PR TITLE
Add transaction list option to get_block_info

### LINKED ISSUE #24: Enhance `get_block_info` tool to optionally include transaction hashes
The `get_block_info` tool currently provides detailed metadata about a specific block but lacks the ability to list the transactions contained within it. To get the transactions, a user would need to make a separate, currently unavailable, call. This enhancement adds an optional feature to `get_block_info` to include a list of transaction hashes directly in its output, improving its utility for block analysis.

### Implementation Details

To address this, the following changes have been implemented:

1.  **`get_block_info` Tool Enhancement:**
    *   The function signature in `blockscout_mcp_server/tools/block_tools.py` has been updated to accept a new optional boolean parameter: `include_transactions: bool = False`.
    *   To maintain backward compatibility and avoid unnecessary API calls, this parameter defaults to `False`.
    *   The function's return type has been changed from `Dict` to `str` to support a more descriptive, multi-part output.

2.  **Conditional Logic:**
    *   If `include_transactions` is `True`, the tool now makes two concurrent API calls using `asyncio.gather`:
        1.  To `/api/v2/blocks/{number_or_hash}` for the block details.
        2.  To `/api/v2/blocks/{number_or_hash}/transactions` for the transaction list.
    *   The transaction list is parsed to extract only the `hash` of each transaction.

3.  **New Output Format:**
    *   The tool now returns a formatted string. When `include_transactions` is `True`, the output is structured as follows:

      ```
      Basic block info:
      {
        "height": 19000000,
        ...
      }

      Transactions in the block:
      [
        "0xtxhash1...",
        "0xtxhash2..."
      ]
      ```
    *   If a block contains no transactions, the second part of the output will be: `No transactions in the block.`
    *   If `include_transactions` is `False`, only the `Basic block info:` section is returned.

### Testing

Comprehensive tests have been added to validate the new functionality:

*   **Unit Tests (`tests/tools/test_block_tools.py`):**
    *   Existing tests were updated to assert against the new string-based output.
    *   New tests were added to cover scenarios with `include_transactions=True`, including blocks with and without transactions, and handling of API errors during the transaction fetch.

*   **Integration Tests (`tests/integration/test_block_tools_integration.py`):**
    *   A new test was added to validate the `include_transactions=True` flag against a stable historical block with a known number of transactions.
    *   Another integration test validates the behavior for a block known to have zero transactions.

### Documentation Updates

The following documentation has been updated to reflect the changes:

1.  **Tool Docstring (`blockscout_mcp_server/tools/block_tools.py`):**
    *   The docstring for `get_block_info` has been updated to describe the new `include_transactions` parameter and advise that it should be used sparingly to avoid extra API load.

2.  **README (`README.md`):**
    *   The description for `get_block_info` in the "Tool Descriptions" section has been updated to mention its new capability.

3.  **Project Structure (`AGENTS.md`):**
    *   The description for `block_tools.py` has been updated to reflect the new functionality of the `get_block_info` tool.

### PR DESCRIPTION
**Pull request by akolotov**
Closes #24 

## Summary
- extend `get_block_info` with `include_transactions` parameter
- support optional concurrent fetch of block transactions
- update unit and integration tests for new behaviour
- document new parameter in README and AGENTS

## Testing
- `pytest`
- `pytest -m integration`

------
https://chatgpt.com/codex/tasks/task_b_684ac35c80648323b2bc5712041c6ac7

### CODE REVIEW

File: blockscout_mcp_server/tools/block_tools.py [hunk 1]
```diff
@@ -1,32 +1,78 @@
...
 from blockscout_mcp_server.tools.common import make_blockscout_request, get_blockscout_base_url
 from mcp.server.fastmcp import Context
 
 async def get_block_info(
     chain_id: Annotated[str, Field(description="The ID of the blockchain")],
     number_or_hash: Annotated[str, Field(description="Block number or hash")],
-    ctx: Context
-) -> Dict:
+    ctx: Context,
+    include_transactions: Annotated[Optional[bool], Field(description="If true, includes a list of transaction hashes from the block. Use this option sparingly as it requires an extra API call.")] = False
```
Comment (akolotov) [inline blockscout_mcp_server/tools/block_tools.py:12]:
```suggestion
    include_transactions: Annotated[Optional[bool], Field(description="If true, includes a list of transaction hashes from the block.")] = False
```

File: blockscout_mcp_server/tools/block_tools.py [hunk 2]
```diff
@@ -1,32 +1,78 @@
...
+) -> str:
     """
     Get block information like timestamp, gas used, burnt fees, transaction count etc.
+    Can optionally include the list of transaction hashes contained in the block. This must be requested only when getting block transactions is really needed.
```
Comment (akolotov) [inline blockscout_mcp_server/tools/block_tools.py:16]:
```suggestion
    Can optionally include the list of transaction hashes contained in the block. Transaction hashes are omitted by default; request them only when you truly need them, because on high-traffic chains the list may exhaust the context.
```
